src/CommandingOfficers/Commander.java

```diff
@@ -567,12 +357,16 @@ public int getIncomePerTurn()
     return count * (gameRules.incomePerCity + incomeAdjustment);
   }
 
-  private final ArrayList<UnitModifier> unitMods;
+  private final ArrayList<UnitModifier> unitMods = new ArrayList<UnitModifier>();
   @Override
   public List<UnitModifier> getModifiers()
   {
-    // TODO Add call to pull modifiers from Army when that becomes a thing?
-    return new ArrayList<UnitModifier>(unitMods);
+    // Intended order of operations: model, D2D, environment, abilities, unit-specific
+    ArrayList<UnitModifier> output = new ArrayList<UnitModifier>();
+    output.add(this);
+    output.addAll(unitMods);
+    output.addAll(this.army.getModifiers());
```

Comment:
I tried starting this game with Team Merge on and got a NPE pointing to this line.
![Screenshot from 2022-01-21 16-39-14](https://user-images.githubusercontent.com/16439262/150609595-f951fd24-41b7-4566-a31e-295c12ff62f3.png)
Text:
```java
Exception in thread "AWT-EventQueue-0" java.lang.NullPointerException
	at CommandingOfficers.Commander.getModifiers(Commander.java:368)
	at Units.UnitContext.<init>(UnitContext.java:103)
	at CommandingOfficers.Commander.getCostContext(Commander.java:204)
	at CommandingOfficers.Commander.getCost(Commander.java:212)
	at CommandingOfficers.Meridian.<init>(Meridian.java:77)
	at CommandingOfficers.Meridian$instantiator.create(Meridian.java:58)
	at UI.PlayerSetupInfo.makeCommander(PlayerSetupInfo.java:77)
	at UI.GameBuilder.createGame(GameBuilder.java:51)
	at UI.PlayerSetupController.handlePlayerSetupInput(PlayerSetupController.java:126)
	at UI.PlayerSetupController.handleInput(PlayerSetupController.java:86)
	at UI.GameOptionSetupController.handleInput(GameOptionSetupController.java:52)
	at UI.MapSelectController.handleInput(MapSelectController.java:37)
	at UI.MainUIController.handleInput(MainUIController.java:65)
	at Engine.Driver.keyPressed(Driver.java:115)
	at java.desktop/java.awt.Component.processKeyEvent(Component.java:6590)
... off into the bowels of awt
```

Reply:
Actually, it worked with all humans as well, which maybe means one of the COs is broken?

Reply:
I ran into this while testing on one of the other later branches.
This is the relevant diff:
```@@ -363,7 +365,10 @@ public class Commander implements GameEventListener, Serializable, UnitModifierW
     ArrayList<UnitModifier> output = new ArrayList<UnitModifier>();
     output.add(this);
     output.addAll(unitMods);
-    output.addAll(this.army.getModifiers());
+    if( null != army )
+      output.addAll(army.getModifiers());
+    else
+      System.out.println("WARNING!: Getting Commander modifiers when CO.army == null!");
     return output;
   }
```
I could throw that in here, but I don't want to develop merge conflicts.
Maybe when we're happy with your current comments here and the upstream branches, we just merge everything down onto this for any final adjustments?

Reply:
Yeah, that's a good plan.

Reply:
Sounds good. And sorry.